src/main.cpp

```diff
@@ -134,6 +136,33 @@ void setup() {
   last_comms = millis();
 }
 
+#define DEBUG_LOG_BUFFER_SIZE 4000
+static char   message_buffer[DEBUG_LOG_BUFFER_SIZE];
+static size_t message_buffer_position;
+int DEBUG_LOG_IMPL(const char *format, ...)
+{
+   int ret;
+   if (message_buffer_position>=DEBUG_LOG_BUFFER_SIZE)
+       return 0;
+   
+   va_list aptr;
+   va_start(aptr, format);
+   ret = vsnprintf(&message_buffer[message_buffer_position], DEBUG_LOG_BUFFER_SIZE-message_buffer_position, format, aptr);
+   va_end(aptr);
+   message_buffer_position+=ret;
+   if (message_buffer_position >= DEBUG_LOG_BUFFER_SIZE) {
+     message_buffer_position=DEBUG_LOG_BUFFER_SIZE;
+   }
+   message_buffer[DEBUG_LOG_BUFFER_SIZE-1]=0;
+
+   return ret;
+}
```

Comment:
Can you please move the logging implementation into a separate file?

Reply:
I removed it; I needed the logging to debug, but it can/should stay local for now